### DISCUSSION TITLE
Implement LQG controller on Linear Goland wing model with Atmospheric turbulence

### DISCUSSION THREAD

**Discussion by xxyao-95** [ImperialCollegeLondon/sharpy#152] [Q&A] [answered]
Hi,

I am trying to design a lqg controller for load allevation of goland wing under atmospheric turbulence.
I have extracted the A B C D matrices from the goland wing example (The state space system has undergo Krylov order reduction).  I have some questions on how to pass the atmospheric turbulence on to the state vector x.
In the state space formulation of LQG controller:
x^dot = A*x + B*u + G*w
G is the covariance matrix and w is the noise. I want to manipulate the G matrix such that atmospheric turbulence can be added on to the states in x and I am struggling with how to find exactly which entry in the x vector to add the disturbance.

From the report, https://arc.aiaa.org/doi/abs/10.2514/6.2021-1798, after Krylov ROM, the u vector is [q_ext, q^dot_ext, w_g, delta, delta^dot and Q_ext], can I conclude that the input u vector already have the gust velocity w_g taken into account? Therefore, if I put atmospheric turbulence into the repective entry of the U vector, the B matrix will apply it accordinly to the states in x, Is this assumption correct?

**Comment by ngoiz** [comment, marked as answer]
First thing just let ask whether you have set up your linear system with these two settings for `settings['LinearAssembler']['aero_settings']` (i..e the settings for the [LinearUVLM](https://ic-sharpy.readthedocs.io/en/master/includes/linear/assembler/linearuvlm/LinearUVLM.html)
* `remove_inputs = []`  or not specified
* `gust_assembler = 'leading_edge'` This will condense all the external velocity inputs (3 x each lattice vertex) to a single vertical velocity input defined at the leading edge. I.e. the velocity you specify for this input will propagate uniformly spanwise and will convect downstream at the free stream velocity.

If you've done that and then reduced your system you are on the right track. Then your system will have as inputs the ones you have already identified and `w_g` will be of dimension 1. The B matrix will indeed propagate through that input as required to the rest of the lattice vertices.

Hope this helps

**Reply by xxyao-95** [reply]
Hi, thank you so much for your reply.
I noticed that I have not specify the gust_assembler settings in my previous extracted state space matrices. Now I have re-extracted the matrices and x change from 35 to 41 states, u change from 8 input to 9 input

For w_g being dimention 1, and "This will condense all the external velocity inputs (3 x each lattice vertex) to a single vertical velocity input defined at the leading edge." does it suggest that I can only put in atmospheric tubulence in the vertical direction as input?

I also want to ask that is it possible to manipulate C matrix to observe the state that I want to observe, for example if I want to observe the tip deflection due to atmospheric turbulence, is it possible to extract that from the state vector x?

**Reply by ngoiz** [reply]
Yes, so the state increases in size to account for the convection of the disturbance downstream. The idea behind the condensation of the inputs is to reduce the number of transfer functions since that is what the ROM is based on (i.e. if the number of inputs/outputs is large, regardless of the state size the resulting reduced order model is going to be of significant size).

That aside, short answer is yes, only a spanwise-constant vertical velocity is supported (provided you want to keep the ROM).

As for the output manipulation, yes, you can extract the wing tip deflection. You can set the `settings['LinearAssembler']['inout_coordinates'] = 'nodes'` and then your system will be expressed in nodal coordinates (eta, eta_dot, w_g, delta..., nodal_forces). Each of these vectors is size `6x(n_nodes - 1)` (each node has 6 degrees of freedom bar the clamped node). All of these are expressed in the beams `A` frame of reference. It would then be a matter of choosing your desired output channel. A good way is to visualise the beam in Paraview and see what is your wing tip node number and then counting to see what is the corresponding output channel for the displacement of said node.

**Reply by xxyao-95** [reply]
Thank you for your reply.

I want to ask that is Krylov Model Order Reduction applicable to nodal coordinates? From the report https://arc.aiaa.org/doi/abs/10.2514/6.2021-1798, it shows that ROM is done in modal coordinate if I understand correctly.

Therefore if I want to extract the wing tip deflection, does that mean my state space is in full state(no ROM)? Since I am designing a LQG controller, more state would make it computationally heavy to calculate the gains.

**Reply by ngoiz** [reply]
You can still use the ROM. The states are going to be unchanged as this affects only the input and output coordinates (it effectively pre and post multiplies the coupled aeroelastic system by the matrix of eigenvectors to project one coordinate set onto the other.

**Reply by xxyao-95** [reply]
Thank you so much for your help! I understand what to do now !
